backend/src/services/company.service.ts

```diff
@@ -0,0 +1,72 @@
+import { getCustomRepository } from 'typeorm';
+import { Company } from '../data/entities/company';
+import { HttpCode, HttpError } from 'growup-shared';
+import { signToken } from '~/common/utils/token.util';
+import { CompanyResponse } from '~/common/models/responses/company';
+
+import CompanyRepository from '~/data/repositories/company.repository';
+
+export const createCompany = async (
+  body: Company,
+): Promise<CompanyResponse> => {
+  const { name } = body;
+
+  const companyRepository = getCustomRepository(CompanyRepository);
+
+  if (name) {
+    const isCompanyExist = await companyRepository.findOne({ name });
+
+    if (!isCompanyExist) {
+      const company = new Company();
+      const newCompany = Object.assign(company, body);
+
+      await newCompany.save();
+
+      const token = signToken(newCompany);
```

Comment:
```suggestion
      const token = signToken({ userId, role, companyId });
```
`userId` and `role` you can pass to service from `req` object.  https://github.com/BinaryStudioAcademy/bsa-winter-2022-growup/blob/develop/backend/src/api/middlewares/authorization-middleware.ts#L25